File: Java/Constructor/Overloaded or copy constructor.java
class constructor {
    constructor(String department) {
        System.out.println("Dept of. " + department);
    }

    constructor(String name, int id) {
        System.out.println("Sudent Name: " + name);
        System.out.println("ID: " + id);
    }
}

class run_constructor {
    public static void main(String[] args) {
        constructor department = new constructor("Computer Engineering");
        constructor student = new constructor("Abdullah Al Naiem", 123456);
    }
}

File: C++/Constructor/Overloaded or copy constructor.cpp
#include <iostream>
using namespace std;

class constructor {
    public:
        constructor(string department){
            cout << "Dept. of " << department << endl;
        }
        constructor(string name, int id){
            cout << "Student Name: " << name << endl;
            cout << "ID: " << id << endl;
        }
};

int main(){
    constructor department("Computer Engineering");
    constructor student("Abdullah Al Naiem", 123456);
}